joblib/testing.py

```diff
@@ -21,7 +21,7 @@
 assert_false = _dummy.assertFalse
 assert_equal = _dummy.assertEqual
 assert_not_equal = _dummy.assertNotEqual
-assert_raises = _dummy.assertRaises
+assert_raises = pytest.raises
```

Comment:
Why not leave `_dummy.assertRaises`? They do not behave exactly the same (pytest.raises has more features) so I would rather have another name for pytest.raises  like:

```py
assert_raises = _dummy.raises
pytest_raises = pytest.raises
```

I don't think we rely on pytest.raises anywhere yet, so I would say leave it as it was previously.